### PR TITLE
247 create folders using empty file

### PR DESCRIPTION
**Pull request by pinussilvestrus**

### CODE REVIEW

**Comment by pinussilvestrus** [comment]
[Create Folders using empty file](https://trello.com/c/Mx72upoe/247-create-folders-using-empty-file)

File: src/modules/file-explorer/actions/file-explorer.js [hunk 1]
```diff
@@ -48,13 +47,33 @@ export default {
...
+			let fakeFile = res.data;
+			return FileService.getUrl(".scfake.png", fakeFile.type, `${storageContext}/${dirName}`, 'putObject')
+			 .then((signedUrl) => {
+				var options = {
+					headers: signedUrl.header
+				};
+
+				return axios.put(signedUrl.url, fakeFile, options);
+			}).catch(err => Notification.showError(err));
+		});
```
Comment (MasterCarl) [inline src/modules/file-explorer/actions/file-explorer.js]: This is an AWS-specific hack (to create a folder by uploading a random file). We shouldn't have this in the client.

File: src/modules/file-explorer/components/file-explorer.jsx [hunk 2]
```diff
@@ -13,6 +13,44 @@ class FileExplorer extends React.Component {
...
+		});
+
+		// delete duplicates in directories
+		let withoutDuplicates = [];
+		directories.forEach(d => {
+			if (withoutDuplicates.indexOf(d) == -1) withoutDuplicates.push(d);
+		});
+
+		// remove .scfake.png fake file
+		files = files.filter(f => f.name != ".scfake.png");
```
Comment (MasterCarl) [inline src/modules/file-explorer/components/file-explorer.jsx]: Same as above

File: src/modules/file-explorer/components/file-explorer.jsx [hunk 3]
```diff
@@ -13,6 +13,44 @@ class FileExplorer extends React.Component {
 		super(props);
 	}
 
+	/**
+	 * split files-list in files, that are in current directory, and the sub-directories
+	 * @param data is the files-list
+     */
+	splitFilesAndDirectories(data) {
```
Comment (MasterCarl) [inline src/modules/file-explorer/components/file-explorer.jsx]: Maybe we should let the server figure this out

Comment (MasterCarl) [inline src/modules/file-explorer/components/file-explorer.jsx]: The current "folder" content without subdirectories can be fetched in S3 by using a `/` delimiter (see <https://forums.aws.amazon.com/thread.jspa?messageID=465624>)

File: src/modules/file-explorer/components/files.jsx [hunk 4]
```diff
@@ -1,18 +1,31 @@
...
 		super(props);
 	}
 
+	getStorageTitle(storageContext) {
+		var values = storageContext.split("/");
+		switch (values[0]) {
+			case 'users': return `Meine persönlichen Dateien /${values.filter((v, index) => {
+					return index > 1;
+				}).join("/")}`;
+			case 'courses': return "Kurs Dateien";
```
Comment (MasterCarl) [inline src/modules/file-explorer/components/files.jsx]: Maybe add a hyphen (**Kurs-Dateien**)

File: src/modules/file-explorer/components/files.jsx [hunk 5]
```diff
@@ -1,18 +1,31 @@
...
 	constructor(props) {
 		super(props);
 	}
 
+	getStorageTitle(storageContext) {
+		var values = storageContext.split("/");
+		switch (values[0]) {
+			case 'users': return `Meine persönlichen Dateien /${values.filter((v, index) => {
+					return index > 1;
+				}).join("/")}`;
```
Comment (MasterCarl) [inline src/modules/file-explorer/components/files.jsx]: It's kind of hard to read this with a function inside a template string

**Pull request merged**